multinode-demo/start_nodes.sh

```diff
@@ -68,40 +66,60 @@ for ip_addr in "${ip_addr_array[@]}"; do
-    ssh -n -f "$remote_user@$ip_addr" "rsync -vPrz ""$remote_user@$leader"":~/solana/multinode-demo ~/solana/"
-    ssh -n -f "$remote_user@$ip_addr" "rsync -vPrz ""$remote_user@$leader"":~/solana/fetch-perf-libs.sh ~/solana/"
-  else
-    # Deploy build and scripts to remote node
-    rsync -vPrz ~/.cargo/bin/solana* "$remote_user@$ip_addr":~/.cargo/bin/
-    rsync -vPrz ./multinode-demo "$remote_user@$ip_addr":~/solana/
-    rsync -vPrz ./fetch-perf-libs.sh "$remote_user@$ip_addr":~/solana/
-  fi
+  rsync -vPrz ~/.cargo/bin/solana* "$remote_user@$ip_addr":~/.cargo/bin/ # Deploy build and scripts to remote node
+  rsync -vPrz ./multinode-demo "$remote_user@$ip_addr":~/solana/
+  rsync -vPrz ./fetch-perf-libs.sh "$remote_user@$ip_addr":~/solana/
 
   # Run setup
   ssh "$remote_user@$ip_addr" "$ssh_command_prefix"' ./multinode-demo/setup.sh -p "$ip_addr"'
 
+  echo "Starting leader node $ip_addr"
+  ssh -n -f "$remote_user@$ip_addr" 'cd solana; ./fetch-perf-libs.sh'
+  ssh -n -f "$remote_user@$ip_addr" "$ssh_command_prefix"' SOLANA_CUDA=1 ./multinode-demo/leader.sh > leader.log 2>&1'
+  ssh -n -f "$remote_user@$ip_addr" "$ssh_command_prefix"' ./multinode-demo/drone.sh > drone.log 2>&1'
+  leader_ip=${ip_addr_array[0]}
+}
+
+validator() {
+  common_setup "$1"
+
+  echo "Adding known hosts for $ip_addr"
+  ssh "$remote_user@$ip_addr" "ssh-keygen -R ""$leader_ip"
+  ssh "$remote_user@$ip_addr" "ssh-keyscan ""$leader_ip >> ~/.ssh/known_hosts"
+
+  ssh "$remote_user@$ip_addr" "rsync -vPrz ""$remote_user@$leader_ip"":~/.cargo/bin/solana* ~/.cargo/bin/"
+  ssh "$remote_user@$ip_addr" "rsync -vPrz ""$remote_user@$leader_ip"":~/solana/multinode-demo ~/solana/"
+  ssh "$remote_user@$ip_addr" "rsync -vPrz ""$remote_user@$leader_ip"":~/solana/fetch-perf-libs.sh ~/solana/"
+
+  # Run setup
+  ssh "$remote_user@$ip_addr" "$ssh_command_prefix"' ./multinode-demo/setup.sh -p "$ip_addr"'
+
+  echo "Starting validator node $ip_addr"
+  ssh -n -f "$remote_user@$ip_addr" "$ssh_command_prefix"" ./multinode-demo/validator.sh $remote_user@$leader_ip:~/solana $leader_ip > validator.log 2>&1"
+}
+
+for ip_addr in "${ip_addr_array[@]}"; do
+  echo "$ip_addr"
+  ssh-keygen -R "$ip_addr"
+  ssh-keyscan "$ip_addr" >>~/.ssh/known_hosts
+
   if ((!count)); then
     # Start the leader on the first node
-    echo "Starting leader node $ip_addr"
-    ssh -n -f "$remote_user@$ip_addr" 'cd solana; ./fetch-perf-libs.sh'
-    ssh -n -f "$remote_user@$ip_addr" "$ssh_command_prefix"' SOLANA_CUDA=1 ./multinode-demo/leader.sh > leader.log 2>&1'
-    ssh -n -f "$remote_user@$ip_addr" "$ssh_command_prefix"' ./multinode-demo/drone.sh > drone.log 2>&1'
-    leader=${ip_addr_array[0]}
+    leader "$ip_addr"
   else
     # Start validator on all other nodes
-    echo "Starting validator node $ip_addr"
-    ssh -n -f "$remote_user@$ip_addr" "$ssh_command_prefix"" ./multinode-demo/validator.sh $remote_user@$leader:~/solana $leader > validator.log 2>&1"
+    validator "$ip_addr" &
   fi
 
```

Comment:
Please add a `sleep` to avoid triggering our login quota (until we can get that bumped up significantly):
https://github.com/solana-labs/solana/blob/master/ci/refresh-testnet.sh#L75-L77

Reply:
Doing it now.. Is sleep 2 absolute minimum? It'll slow down overall startup time..

Reply:
Right now we get 60 OS logins per minute before GCP starts rejecting `ssh` commands, and it looks like the current script `ssh`s 4-5 times per machine, so we'll hit that 60 per minute limit very quickly otherwise